src/main/java/software/amazon/cloudformation/resource/Serializer.java

```diff
@@ -76,10 +87,37 @@
         return OBJECT_MAPPER.writeValueAsString(modelObject);
     }
 
+    public <T> String compress(final String modelInput) throws IOException {
+        final Map<String, String> map = new HashMap<>();
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        GZIPOutputStream gzip = new GZIPOutputStream(byteArrayOutputStream);
+        gzip.write(modelInput.getBytes(StandardCharsets.UTF_8));
+        gzip.close();
+        map.put(COMPRESSED, Base64.encodeBase64String(byteArrayOutputStream.toByteArray()));
+        return OBJECT_MAPPER.writeValueAsString(map);
+    }
+
     public <T> T deserialize(final String s, final TypeReference<T> reference) throws IOException {
         return OBJECT_MAPPER.readValue(s, reference);
     }
 
+    public String decompress(final String s) {
+        try {
+            final Map<String, String> map = deserialize(s, MAP_TYPE_REFERENCE);
+
+            if (!map.containsKey(COMPRESSED)) {
+                return s;
+            }
+
+            final byte[] bytes = Base64.decodeBase64(map.get(COMPRESSED));
+            ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(bytes);
+            GZIPInputStream gzipInputStream = new GZIPInputStream(byteArrayInputStream);
+            return new String(IOUtils.toByteArray(gzipInputStream), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            return s;
```

Comment:
I don't see a reason why it would fail. But as least add logging here for the exception.